apps/web/src/hooks/useRenewBasename.ts

```
@@ -0,0 +1,116 @@
+  const { initiateBatchCalls, batchCallsStatus, batchCallsIsLoading, batchCallsError } =
+    useWriteContractsWithLogs({
+      chain: basenameChain,
+      eventName: 'renew_name',
+    });
+
+  // Transaction without paymaster
+  const {
+    initiateTransaction: initiateRenewName,
+    transactionStatus: renewNameStatus,
+    transactionIsLoading: renewNameIsLoading,
+    transactionError: renewNameError,
+  } = useWriteContractWithReceipt<typeof REGISTER_CONTRACT_ABI, 'renew'>({
+    chain: basenameChain,
+    eventName: 'renew_name',
+  });
+
+  // Params
+  const normalizedName = normalizeEnsDomainName(name);
+  const { basePrice, premiumPrice } = useRentPrice(normalizedName, years);
+  const totalPrice = basePrice + premiumPrice;
+
+  // Callback
+  const renewName = useCallback(async () => {
+    if (!address) {
+      return;
+    }
+
+    const renewRequest = [normalizedName, secondsInYears(years)];
+
+    try {
+      if (!paymasterServiceEnabled) {
+        console.log('Renewing name without paymaster', {
+          renewRequest,
+          totalPrice,
+          contractAddress: REGISTER_CONTRACT_ADDRESSES[basenameChain.id],
+        });
+        await initiateRenewName({
+          abi: REGISTER_CONTRACT_ABI,
+          address: REGISTER_CONTRACT_ADDRESSES[basenameChain.id],
+          functionName: 'renew',
+          args: renewRequest,
+          value: totalPrice,
+        });
+      } else {
+        console.log('Renewing name with paymaster');
+        await initiateBatchCalls({
+          contracts: [
+            {
+              abi: REGISTER_CONTRACT_ABI,
+              address: REGISTER_CONTRACT_ADDRESSES[basenameChain.id],
+              functionName: 'renew',
+              args: renewRequest,
+              value: totalPrice,
+            },
+          ],
+          account: address,
+          chain: basenameChain,
+        });
+      }
+    } catch (e) {
+      logError(e, 'Renew name transaction canceled');
```

Should we throw the error after logging it?